openfisca_france/model/prelevements_obligatoires/impot_revenu/ir.py

```diff
@@ -2712,18 +2786,19 @@ def function(self, simulation, period):
         veuf = simulation.calculate('veuf', period)
         celibataire_ou_divorce = simulation.calculate('celibataire_ou_divorce', period)
         nbptr = simulation.calculate('nbptr', period)
-        ppe = simulation.legislation_at(period.start).ir.credits_impot.ppe
+        ppe = simulation.legislation_at(period.start).impot_revenu.credits_impot.ppe
 
         seuil = (veuf | celibataire_ou_divorce) * (ppe.eligi1 + 2 * max_(nbptr - 1, 0) * ppe.eligi3) \
                 + maries_ou_pacses * (ppe.eligi2 + 2 * max_(nbptr - 2, 0) * ppe.eligi3)
         return period, (rfr * ppe_coef) <= seuil
 
 
-class ppe_rev(Variable):
+class ppe_rev(DatedVariable):
     column = FloatCol
     entity_class = Individus
     label = u"ppe_rev"
 
+    @dated_function(stop = date(2015, 12, 31))
```

Comment:
This is a matter of choice, no? Whatever, #547 will merge these two approaches to one and we'll convert the source code to the new way.